### PR TITLE
Add creating of concept exercises PRs via API

### PR DESCRIPTION
**Pull request by iHiD**
Ensure:
```
rake db:schema:load
rake db:seed
User.first.confirm
```
You need a token from `User.first.auth_token`

This is currently the minimal working cmd:

```
wget --header='AUTHORIZATION: Token token=747f80c0-213a-4c4c-9a01-0ceef827770b' --post-data 'exercise_slug=bob&example_filename=bob.rb' 'http://lvh.me:3000/api/maintaining/concept_exercises'
```

### TODO

- [ ] Check all params are present
- [ ] git setup script
- [ ] git commit script

### CODE REVIEW

File: app/controllers/api/maintaining/concept_exercises_controller.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  module Maintaining
+    class ConceptExercisesController < BaseController
+      def create
+        example_filename_regexp = "[^a-zA-Z0-9._-]"
+        exercise_slug_regexp = "[^a-zA-Z0-9-]"
+
+        unless params[:exercise_slug].present?
+          return render_error(
+            :missing_exercise_slug,
+            "Exercise Slug is missing"
```
Comment (ErikSchierboom) [inline app/controllers/api/maintaining/concept_exercises_controller.rb:11]: For consistency with the later error message:

```suggestion
            "Exercise slug is missing"
```

File: app/controllers/api/maintaining/concept_exercises_controller.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        end
+
+        uuid = SecureRandom.uuid
+        tmp_path = Pathname.new("/tmp/#{uuid}")
+        branch_name = "#{params[:exercise_slug]}-#{uuid}"
+
+        p tmp_path
+
+        #exec("ce-setup.sh", tmp_path)
+        Dir.mkdir tmp_path # TODO Git checkout should do this.
```
Comment (iHiD) [inline app/controllers/api/maintaining/concept_exercises_controller.rb:44]: Erik, so this should:
```
git clone v3-url $tmp_path
cd $tmp_path
git remote add s3-wizard-url
git checkout -b $exercise_name-$some_uuid
```

File: app/controllers/api/maintaining/concept_exercises_controller.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+        File.open(tmp_path / "introduction.md", "w") {|f|f.write params[:introduction_markdown] }
+        File.open(tmp_path / "design.md", "w") {|f|f.write params[:design_markdown] }
+        File.open(tmp_path / params[:example_filename], "w") {|f|f.write params[:example_code] }
+
+        #exec("ce-commit.sh", tmp_path)
```
Comment (iHiD) [inline app/controllers/api/maintaining/concept_exercises_controller.rb:50]: And this should:
```
cd $tmp_path
git add .
git commit -am "[$lang_name] Add $exercise_name"
git push wizard HEAD
```

File: app/controllers/api/maintaining/concept_exercises_controller.rb (hunk 1 above)
Comment (ErikSchierboom) [reply app/controllers/api/maintaining/concept_exercises_controller.rb:11]: Would it make sense to list the exact API field that is missing? Something like:

```suggestion
            "Exercise Slug is missing. Please post it in the 'missing_exercise_slug' field."
```

Or something similar.

Comment (iHiD) [reply app/controllers/api/maintaining/concept_exercises_controller.rb:11]: Maybe yeah. It depends on whether you want to put the errors here or in the client. I'd potentially delete the message altogether here, and just maintain the text in the client (based on the "type" the API returns) as it probably needs to correspond to the UI, not the API.

File: app/controllers/api/maintaining/concept_exercises_controller.rb [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+        uuid = SecureRandom.uuid
+        tmp_path = Pathname.new("/tmp/#{uuid}")
+        branch_name = "#{params[:exercise_track]}/#{params[:exercise_slug]}-#{uuid}"
+
+        p tmp_path
+
+        `git clone https://#{Rails.application.secrets.exercism_bot_credentials}@github.com/exercism-bot/v3.git #{tmp_path}`
+
+        Dir.chdir(tmp_path) {
```
Comment (iHiD) [inline app/controllers/api/maintaining/concept_exercises_controller.rb:53]: Use `do`/`end` for multiline blocks that don't have some method tagged onto the end:

These are the "valid" approaches in my opinion (ie the opinion of our Ruby codebases :))

```ruby
[].each do
# ...
end

[].each { # ... }

[].each { 
}.something_else
```

Comment (ErikSchierboom) [reply app/controllers/api/maintaining/concept_exercises_controller.rb]: Right! Makes sense. I've updated

**Pull request merged**